### PR TITLE
Refs #10755 - fix setting time in the future (was defauling to Time.now)

### PR DESCRIPTION
**Pull request by iNecas**
Also, add attr_accessilble to work better with rails 4.

### CODE REVIEW

**Comment by iNecas** [comment]
[test]

**Comment by iNecas** [comment]
@adamruzicka could you review and test this one? I would like to get it in before tomorrows release

File: app/models/foreman_tasks/triggering.rb [hunk 1]
```diff
@@ -39,11 +40,11 @@ class Triggering < ActiveRecord::Base
     validate :correct_cronline, :if => Proc.new { |t| t.recurring? && t.input_type == :cronline }
 
     def self.new_from_params(params = {})
-      self.new(params).tap do |targeting|
+      self.new(params.except(:mode)).tap do |targeting|
         targeting.mode = params.fetch(:mode, :immediate).to_sym
         targeting.input_type = params.fetch(:input_type, :daily).to_sym
         targeting.end_time_limited = params[:end_time_limited] == "true"
-        targeting.start_at_raw = Time.now.strftime(TIME_FORMAT)
+        targeting.start_at_raw ||= Time.now.strftime(TIME_FORMAT)
```
Comment (adamruzicka) [inline app/models/foreman_tasks/triggering.rb]: Could you please `s/targeting/triggering` as part of this pr? Don't know why I named it targeting when it's triggering

**Comment by adamruzicka** [comment]
Code-wise looks good, tests will have to wait till tomorrow

**Comment by adamruzicka** [comment]
Works as expected, ACK

File: app/models/foreman_tasks/triggering.rb [hunk 2]
```diff
@@ -39,11 +40,11 @@ class Triggering < ActiveRecord::Base
     validate :correct_cronline, :if => Proc.new { |t| t.recurring? && t.input_type == :cronline }
 
     def self.new_from_params(params = {})
-      self.new(params).tap do |targeting|
+      self.new(params.except(:mode)).tap do |targeting|
```
Comment (adamruzicka) [inline app/models/foreman_tasks/triggering.rb]: When trying to rerun job I'm getting `Can't mass-assign protected attributes: start_at, start_before`
`params.except(:mode, :start_at, :start_before)` fixes the issue

**Comment by iNecas** [comment]
@adamruzicka thanks for testing, issues addressed

**Comment by adamruzicka** [comment]
ACK

**Pull request merged**